From eadd3411215d5c76d8ef26bd63b205ca5726da58 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Mon, 10 Feb 2025 13:51:54 -0300
Subject: [PATCH 10/10] giscanner: Consider UCRT64 and CLANG64 as mingw32
 compilers

gobject-introspection's own build doesn't propagate CC, so it is not a reliable
detector. However, other Visual Studio Shell variables are available;
VCINSTALLDIR is a known good one [1].

[1]: https://learn.microsoft.com/en-us/cpp/windows/redistributing-visual-cpp-files?view=msvc-170

---
 giscanner/ccompiler.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/giscanner/ccompiler.py b/giscanner/ccompiler.py
index 9a732cd5..052cb732 100644
--- a/giscanner/ccompiler.py
+++ b/giscanner/ccompiler.py
@@ -150,7 +150,8 @@ class CCompiler(object):
             # as the official Python binaries are built with
             # Visual Studio
             if compiler_name is None:
-                if environ.get('MSYSTEM') == 'MINGW32' or environ.get('MSYSTEM') == 'MINGW64':
+                mingw = environ.get('MSYSTEM')
+                if mingw.startswith(('MINGW', 'CLANG', 'UCRT')) and  environ.get('VCINSTALLDIR') is None:
                     compiler_name = 'mingw32'
                 else:
                     compiler_name = distutils.ccompiler.get_default_compiler()
-- 
2.47.0

